### PR TITLE
Sonar fixes in AnalyzerGuru

### PR DESCRIPTION
**Pull request by vladak**
Another round of Sonar fixes.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java [hunk 1]
```diff
@@ -1158,10 +1156,8 @@ private static String readOpening(InputStream in, byte[] sig)
 
             // If the opening starts with "#!", then track so that any
             // trailing whitespace after the hashbang is ignored.
-            if (opening.length() == 2) {
-                if (opening.charAt(0) == '#' && opening.charAt(1) == '!') {
-                    postHashbang = true;
-                }
+            if (opening.length() == 2 && opening.charAt(0) == '#' && opening.charAt(1) == '!') {
```
Comment (ahornace) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java:1159]: Wouldn't `opening.equals("#!")` be nicer?

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java]: Definitely nicer, thanks !

Comment (ahornace) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java]: My bad, I didn't notice that `opening` is a `StringBuilder`, in that case `toString` operation might be expensive if we execute it for every character.

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java:1158]: ok, reverting

**Pull request merged**